extensions/VisualEditor/ApiVisualEditor.php

```diff
@@ -367,7 +385,12 @@ public function execute() {
 		wfDebugLog( 'visualeditor', "called on '$page' with paction: '{$params['paction']}'" );
 		switch ( $params['paction'] ) {
 			case 'parse':
-				$parsed = $this->getHTML( $page, $parserParams );
+				if ( isset( $params['preload'] )) {
+					$parsed = $this->getPreloadHTML( $params );
```

Comment:
This is an API and VE just happens to be the main client of it - but it
doesn't mean that we should return incorrect error because VE is going to
handle it the same way (creating a ticket to have VE handle it in more
robust way perhaps would be more appropriate action).
The more I look at it the more it looks that API was already good and
changes should be done on the client.
On Feb 21, 2016 22:17, "Per Johan Groland" <notifications@github.com> wrote:

> In extensions/VisualEditor/ApiVisualEditor.php
> <https://github.com/Wikia/app/pull/9710#discussion_r53589465>:
>
> > @@ -367,7 +385,12 @@ public function execute() {
> >  		wfDebugLog( 'visualeditor', "called on '$page' with paction: '{$params['paction']}'" );
> >  		switch ( $params['paction'] ) {
> >  			case 'parse':
> > -				$parsed = $this->getHTML( $page, $parserParams );
> > +				if ( isset( $params['preload'] )) {
> > +					$parsed = $this->getPreloadHTML( $params );
>
> I moved the extraction of the title into the call of getPreloadHTML like
> suggested.
>
> The error you mention will happen if the preload parameter is pointing to
> a page that doesn't exist.
> I tested it (see error section here:
> http://visualeditor.pgroland.wikia-dev.com/wiki/Test1) and it doesn't end
> up with a user-facing error, it just shows empty contents in VE.
>
> I could rewrite the code like so:
>
>             if ( isset( $params['preload'] )) {
>                 $parsed = $this->getPreloadHTML( $urldecode( $params['preload'] ) );
>             }
>
>             if ( !$parsed )
>                 $parsed = $this->getHTML( $page, $parserParams );
>             }
>
> Which might be safer, but I don't see any difference in the user
> experience. Is there something else going on (logging?) that means that
> approach is better?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Wikia/app/pull/9710/files#r53589465>.
>